tinybmp/src/lib.rs

```diff
@@ -184,9 +230,75 @@ impl<'a> Iterator for BmpIterator<'a> {
 
             self.start_idx += self.pixel_stride;
 
-            Some(pixel_value)
+            Some(Pixel {
+                x,
+                y,
+                color: pixel_value,
+            })
         } else {
             None
         }
     }
 }
+
+#[cfg(feature = "graphics")]
+mod e_g {
+    use super::*;
+    use core::marker::PhantomData;
+    use embedded_graphics::{
+        drawable::Pixel as EgPixel,
+        geometry::Point,
+        image::ImageData,
+        pixelcolor::{raw::RawData, PixelColor},
+    };
+
+    /// A thin wrapper over [`BmpIterator`] to support [`embedded-graphics`] integration
+    ///
+    /// [`BmpIterator`]: ./struct.BmpIterator.html
+    /// [`embedded-graphics`]: https://docs.rs/embedded-graphics
+    #[derive(Debug)]
+    pub struct EgPixelIterator<'a, C> {
+        it: BmpIterator<'a>,
+        c: PhantomData<C>,
+    }
+
+    impl<'a, C> Iterator for EgPixelIterator<'a, C>
+    where
+        C: PixelColor + From<<C as PixelColor>::Raw>,
+    {
+        type Item = EgPixel<C>;
+
+        fn next(&mut self) -> Option<Self::Item> {
+            self.it.next().map(|p| {
+                let raw = C::Raw::from_u32(p.color);
+                EgPixel(Point::new(p.x as i32, p.y as i32), raw.into())
+            })
+        }
+    }
+
+    /// TODO: Docs
+    impl<'a, C> ImageData<C> for &'a Bmp<'a>
+    where
+        C: PixelColor + From<<C as PixelColor>::Raw>,
+    {
+        type PixelIterator = EgPixelIterator<'a, C>;
+
+        fn width(&self) -> u32 {
+            self.header.image_width
```

Comment:
If I call `self.width()` here instead I get this weird error, likely getting confused between `ImageData::width` and `Bmp::width`:

```
error[E0283]: type annotations needed: cannot resolve `_: embedded_graphics::pixelcolor::PixelColor`
```

I thought this might happen with the trait methods having the same name as methods on the image. I'm pretty ok just using `self.header.image_width` here, but it feels like a bit of a hack and will confuse anybody adding other image libraries in the future for sure. Any ideas?

Reply:
Ah never mind, figured it out!

```rust
fn width(&self) -> u32 {
    Bmp::width(&self)
}
```

I thought I tried this before but oh well, tried it again and it works now :ok_hand: